File: app/src/net/reichholf/dreamdroid/helpers/enigma2/PowerState.java
/* © 2010 Stephan Reichholf <stephan at reichholf dot net>
 * 
 * Licensed under the Create-Commons Attribution-Noncommercial-Share Alike 3.0 Unported
 * http://creativecommons.org/licenses/by-nc-sa/3.0/
 */

package net.reichholf.dreamdroid.helpers.enigma2;

import androidx.annotation.NonNull;

import net.reichholf.dreamdroid.helpers.NameValuePair;

import java.util.ArrayList;

/**
 * @author sre
 *
 */
public class PowerState {
	@NonNull
	public static String KEY_IN_STANDBY = "standby";
	
	@NonNull
	public static String CMD_SET = "set";
	
	@NonNull
	public static String STATE_GET = "-1";
	@NonNull
	public static String STATE_TOGGLE = "0";
	@NonNull
	public static String STATE_SHUTDOWN = "1";
	@NonNull
	public static String STATE_SYSTEM_REBOOT = "2";
	@NonNull
	public static String STATE_GUI_RESTART = "3";
	
	@NonNull
	public static ArrayList<NameValuePair> getStateParams(String state){
		ArrayList<NameValuePair> params = new ArrayList<>();
		params.add(new NameValuePair("newstate", state) );
		
		return params;
	}
}

File: app/src/net/reichholf/dreamdroid/helpers/enigma2/PowerState.kt
package net.reichholf.dreamdroid.helpers.enigma2

import net.reichholf.dreamdroid.helpers.NameValuePair

object PowerState {
    const val KEY_IN_STANDBY: String = "standby"

    const val CMD_SET: String = "set"

    const val STATE_GET: String = "-1"
    const val STATE_TOGGLE: String = "0"
    const val STATE_SHUTDOWN: String = "1"
    const val STATE_SYSTEM_REBOOT: String = "2"
    const val STATE_GUI_RESTART: String = "3"

    fun getStateParams(state: String?): ArrayList<NameValuePair> {
        val params = ArrayList<NameValuePair>()
        params.add(NameValuePair("newstate", state ?: ""))
        return params
    }
}
